File: euler/euler2/loop.rs
fn sum_filtered_fib(max: u64, div_by: u64) -> u64 {
    let mut res = 0;
    let mut last = 0;
    let mut curr = 1;
    let mut tmp;

    while (curr <= max) {
        if (curr % div_by == 0) {
            res += curr;
        }
        tmp = curr;
        curr += last;
        last = tmp;
    }

    return res;
}

use std::env;
use std::str::FromStr;

fn main() {
    let args: Vec<String> = env::args().collect();
    let args = &args[1..];
    let args: Vec<u64> = args.iter().map(|x| u64::from_str(&x).unwrap()).collect();
    let res = sum_filtered_fib(args[0], args[1]);
    println!("{}", res);
}

File: euler/euler2/loop.c
#include <stdio.h>

long long sum_filtered_fib(long long max, long long div_by) {
    long long res = 0;
    long long last = 0;
    long long curr = 1;
    long long tmp;

    while (curr <= max) {
        if (curr % div_by == 0) {
            res += curr;
        }
        tmp = curr;
        curr += last;
        last = tmp;
    }

    return res;
}

int main(int argc, char const *argv[])
{
	int res = sum_filtered_fib(atoll(argv[1]), atoll(argv[2]));
	printf("%d\n", res);
	return 0;
}
